### DISCUSSION TITLE
Error: If no blocks are specified, predictorMatrix must have same number of rows and columns

### DISCUSSION THREAD

**Discussion by DanielIAvila** [amices/mice#591] [Impute, analyse and pool]
I have a really "big" problem using `mice`. I have posted a question [here](https://stackoverflow.com/questions/77235170/imputation-failing-with-predictormatrix-and-custom-imputation-in-mice).

I am getting an error (displayed in the title), but my `df_a` has `160 rows` and  `6 columns`, where the  `predictor_matrix` has the same shape, except that the first three columns are set to  `FALSE` and the remaining three to  `TRUE`.

**Comment by hanneoberman** [comment]
I believe this is not a bug in `mice`, but a problem with the custom imputation function. The function you wrote does not adhere to the `mice.impute.XYZ` conventions, where `XYZ` refers to your imputation model, see e.g. https://github.com/amices/mice/blob/master/R/mice.impute.mpmm.R for an example of a multivariate imputation method.
